pkg/util/util.go

```diff
@@ -175,6 +175,26 @@ func GetClusterByName(ctx context.Context, c client.Client, namespace, name stri
 	return cluster, nil
 }
 
+// GetMachineOwner returns the Machine object owning the current resource.
+func GetMachineOwner(ctx context.Context, c client.Client, obj metav1.ObjectMeta) (*clusterv1.Machine, error) {
```

Comment:
The name threw me off for a moment, because I was expecting it to return the object referenced by the owner ref on the machine rather than returning a Machine that is referenced by an owner ref on an object.

Would this be better named `GetOwnerMachine` instead?

Reply:
Sure!

Reply:
Fixed